btsieve/src/bitcoin/bitcoind_http_blocksource.rs

```diff
@@ -0,0 +1,205 @@
+        block_hash: String,
+    ) -> impl Future<Item = Block, Error = Error> + Send + 'static {
+        let raw_block_by_hash_url = format!("{}/rest/block/{}.hex", self.base_url, block_hash);
+
+        self.client
+            .get(raw_block_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(move |response_text| {
+                let mut response_text: String = response_text;
+                remove_tailing_linebreak(&mut response_text);
+                hex::decode(response_text).map_err(Error::Hex)
+            })
+            .and_then(|bytes| deserialize(bytes.as_ref()).map_err(Error::BlockDeserialization))
+            .map(move |block| {
+                log::trace!("Got {:?}", block);
+                block
+            })
+    }
+
+    /// Currently needed because we don't store the complete transaction in the
+    /// query result, thus we have to fetch the transaction by ID when the
+    /// result is requested by CND. This problems should be solved by
+    /// storing complete transactions instead of just their ID.
+    pub fn transaction_by_hash(
+        &self,
+        transaction_hash: String,
+    ) -> impl Future<Item = Transaction, Error = Error> + Send + 'static {
+        let raw_transaction_by_hash_url =
+            format!("{}/rest/tx/{}.hex", self.base_url, transaction_hash);
+
+        self.client
+            .get(raw_transaction_by_hash_url.as_str())
+            .send()
+            .map_err(Error::Reqwest)
+            .and_then(|mut response| response.text().map_err(Error::Reqwest))
+            .and_then(move |response_text| {
+                let mut response_text: String = response_text;
+                remove_tailing_linebreak(&mut response_text);
+                hex::decode(response_text).map_err(Error::Hex)
+            })
+            .and_then(|bytes| {
+                deserialize(bytes.as_ref()).map_err(|e| {
+                    log::error!(
+                        "Got new transaction but failed to deserialize it because {:?}",
+                        e
+                    );
+                    Error::TransactionDeserialization(format!(
+                        "Failed to deserialize the response from bitcoin-d into a transaction: {}",
+                        e
+                    ))
+                })
+            })
+            .map(move |transaction| {
+                log::debug!("Fetched transaction {:?}", transaction);
+                transaction
+            })
+    }
+}
+
+impl BlockSource for BitcoindHttpBlockSource {
+    type Block = Block;
+    type Error = Error;
+
+    fn blocks(
+        &self,
+    ) -> Box<dyn Stream<Item = Self::Block, Error = blocksource::Error<Error>> + Send> {
+        // The Bitcoin blockchain has a mining interval of about 10 minutes.
+        // The poll interval is configured to once every 5 minutes for mainnet and
```

Comment:
Which means you may have to wait up to 5min after a block is mined.
What about setting it to 2 minutes to reduce the wait yet not checking too frequently?
Especially that here, there are no quota restrictions.